account/views.py

```diff
@@ -63,11 +83,66 @@ def post(self, request):
                     {"message": "Password is incorrect"},
                     status=status.HTTP_400_BAD_REQUEST,
                 )
-            user_profile = UserProfile.objects.get(user=user)
-            user_profile_serializer = UserProfileSerializer(instance=user_profile)
-            return Response(user_profile_serializer.data, status=status.HTTP_200_OK)
+            
+            return set_token_on_response_cookie(user, status_code=status.HTTP_200_OK)
 
         except User.DoesNotExist:
             return Response(
                 {"message": "User does not exist"}, status=status.HTTP_404_NOT_FOUND
             )
+
+class TokenRefreshView(APIView):
+    @swagger_auto_schema(
+        operation_id="토큰 재발급",
+        operation_description="access 토큰을 재발급 받습니다.",
+        request_body=TokenRefreshRequestSerializer,
+        responses={200: UserProfileSerializer},
+    )
+    def post(self, request):
+        refresh_token = request.data.get("refresh")
+        
+        #### 1
+        if not refresh_token:
+            return Response(
+                {"detail": "no refresh token"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+        #### 2
+            RefreshToken(refresh_token).verify()
+        except:
+            return Response(
+                {"detail": "please signin again."}, status=status.HTTP_401_UNAUTHORIZED
+            )
+            
+        #### 3
+        new_access_token = str(RefreshToken(refresh_token).access_token)
+        response = Response({"detail": "token refreshed"}, status=status.HTTP_200_OK)
+        response.set_cookie("access_token", value=str(new_access_token), httponly=True)
+        return response
+
+class LogoutView(APIView):
+    @swagger_auto_schema(
+        operation_id="로그아웃",
+        operation_description="사용자를 로그아웃 시킵니다.",
+        request_body=TokenRefreshRequestSerializer,
+        responses={204: "", 400: "Bad Request", 401: "UNAUTHORIZED"},
+        manual_parameters=[openapi.Parameter("Authorization", openapi.IN_HEADER, description="access token", type=openapi.TYPE_STRING)]
+    )
+    def post(self, request):
```

Comment:
로그인을 한 사용자만 로그아웃을 할 수 있겠죠..!! refresh token을 검사하기 전에 access token을 바탕으로 이 유저가 로그인되어 있는 상태인지를 검사해 주면 좋을 것 같아요.
```suggestion
    def post(self, request):
        if not request.user.is_authenticated:
            return Response({"detail": "please signin."}, status=status.HTTP_401_UNAUTHORIZED)
```